### PR TITLE
Add move contract

### LINKED ISSUE #422: Add move contract
**Brief of your idea**
Add one new contract, which supports that only the local address can operate this contract that contains transfer method and getBalance

**What could be better if XuperChain using your idea**
Move action

**Are there any side effects?**
Describe what might be worse.

**Additional information**
If possible, include a minimal, complete, and verifiable example to help us identify the issue.

### PR DESCRIPTION
**Pull request by MichelleZ**
## Description

What is the purpose of the change?
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #422 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## Brief of your solution

Please describe your solution to solve the issue or feature request.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### CODE REVIEW

File: contractsdk/cpp/example/move.cc [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+enum ret_t {
+    RET_SUCCESS = 0,  
+    RET_ERROR_INVALID_NUM,
+    RET_ERROR_OVERFLOW
+};
+
+ret_t string2num(const std::string& from, int64_t *to) {
+    long long temp;
+    temp = std::stoll(from.c_str(), NULL, 10);
```
Comment (duanbing) [inline contractsdk/cpp/example/move.cc]: if non-digit char included in "from" like "123abc", 123 will be parsed to number. Better to pass a char *p to a pointer to the first non-digit char.

File: contractsdk/cpp/example/move.cc [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    if (ctx->get_object(to_key, &value)) {
+        if (string2num(value.c_str(), &to_balance) != RET_SUCCESS) {
+            ctx->error("The balance format of to is wrong");
+            return;
+        }
+    }
+  
+    from_balance = from_balance - token;
+    if (LLONG_MAX - to_balance > token) {
+        ctx->error("If to is added the token, his amount will overflow");
```
Comment (duanbing) [inline contractsdk/cpp/example/move.cc]: should return here.

File: contractsdk/cpp/example/move.cc [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    const std::string& token_str = ctx->arg("token");
+    if (token_str.empty()) {
+        ctx->error("missing token");
+        return;
+    }
+    int64_t token = 0;
+    if (string2num(token_str, &token) != RET_SUCCESS) {
+        ctx->error("token is not valid");
```
Comment (duanbing) [inline contractsdk/cpp/example/move.cc]: return.

File: contractsdk/cpp/example/move.cc [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        return;
+    }
+    const std::string& totalSupply = ctx->arg("totalSupply");
+    if (totalSupply.empty()) {
+        ctx->error("missing totalSupply");
+        return;
+    }
+    int64_t total = 0;
+    if (string2num(totalSupply, &total) != RET_SUCCESS) {
+        ctx->error("totalSupply is not valid");
```
Comment (duanbing) [inline contractsdk/cpp/example/move.cc]: return

File: contractsdk/cpp/example/move.cc (hunk 1 above)
Comment (MichelleZ) [reply contractsdk/cpp/example/move.cc]: Yes, you're right. Got it. The pointer will go to the next position of the numerical number. I'll fix it.

File: contractsdk/cpp/example/move.cc (hunk 2 above)
Comment (MichelleZ) [reply contractsdk/cpp/example/move.cc]: Fix it.

File: contractsdk/cpp/example/move.cc (hunk 3 above)
Comment (MichelleZ) [reply contractsdk/cpp/example/move.cc]: Fix it

File: contractsdk/cpp/example/move.cc (hunk 4 above)
Comment (MichelleZ) [reply contractsdk/cpp/example/move.cc]: Fix it.

File: contractsdk/cpp/example/move.cc [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        return;
+    }
+
+    std::string key = BALANCEPRE + caller;
+    ctx->put_object(key, totalSupply);
```
Comment (fxsjy) [inline contractsdk/cpp/example/move.cc]: check the return of pub_object

File: contractsdk/cpp/example/move.cc [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    from_balance = from_balance - token;
+    if (LLONG_MAX - to_balance < token) {
+        ctx->error("If to is added the token, his amount will overflow");
+        return;
+    }
+    to_balance = to_balance + token;
+   
+    char buf[64]; 
+    snprintf(buf, 64, "%lld", from_balance);
+    ctx->put_object(from_key, buf);
```
Comment (fxsjy) [inline contractsdk/cpp/example/move.cc]: check the return

File: contractsdk/cpp/example/move.cc [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    snprintf(buf, 64, "%lld", to_balance);
+    ctx->put_object(to_key, buf);
```
Comment (fxsjy) [inline contractsdk/cpp/example/move.cc]: check the return

**Pull request merged**